spec.go

```diff
@@ -15,6 +16,7 @@ import (
 type Interface interface {
 	CRDClient() k8scrdclient.Interface
 	CtrlClient() client.Client
+	DiscoveryClient() discovery.DiscoveryInterface
```

Comment:
I thought any client provides a discovery client along with the rest client already. Why do we need to add this separately now?  I was also under the impression that the discovery client needs specific encoding information which is exactly the reason why it is part of the more specific client implementations.

Reply:
Oh I'm stupid. Totally forgot about that.